notifier/management/commands/forums_digest_flagged.py

```diff
@@ -0,0 +1,75 @@
+from django.core.management.base import BaseCommand
+from optparse import make_option
+
+from notifier.digest import render_digest_flagged
+from notifier.pull import get_flagged_threads
+from notifier.tasks import do_forums_digests_flagged
+from notifier.user import get_moderators
+
+
+class Command(BaseCommand):
+    """
+    This Command is used to send a digest of threads with flagged posts to
+    forum moderators.
+    """
+
+    help = "Send a digest list of threads with flagged posts to each moderator"
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--is-dry-run',
+            action='store_true',
+            dest='is_dry_run',
+            default=False,
+            help='for each course with flagged posts, display the email address' +
+                 ' for each recipient, and the rendered email text WITHOUT' +
+                 ' sending a message',
+        ),
+    )
+
+    def handle(self, *args, **options):
+        """
+        Handle a request to send a digest of threads with flagged posts to forum
+        moderators.
+        """
+        is_dry_run = options.get('is_dry_run')
+
+        if is_dry_run:
+            course_threads = {}
+            for thread in get_flagged_threads():
+                course_id = thread.get('course_id')
+                commentable_id = thread.get('commentable_id')
+                comment_thread_id = thread.get('comment_thread_id')
+                if course_id and commentable_id and comment_thread_id:
+                    if not course_threads.has_key(course_id):
+                        course_threads[course_id] = []
+                    thread_url = '{}/courses/{}/discussion/forum/{}/threads/{}'.format(
+                        settings.LMS_URL_BASE, course_id, commentable_id, comment_thread_id
+                    )
+                    course_threads[course_id].append(thread_url)
+
+            for course_id in course_threads:
+                moderators = get_moderators(course_id)
+                try:
+                    user = moderators.next()
+                except StopIteration:
+                    print('Course {} has flagged posts but no moderators'.format(course_id))
+                    continue
+                text, html = render_digest_flagged({
+                    'course_id': course_id,
+                    'threads': course_threads[course_id],
+                    'recipient': user,
+                })
+                print "=============================="
+                print("To: {email_address}".format(email_address=user.get('email')))
+                for user in moderators:
+                    print("To: {email_address}".format(
+                        email_address=user.get('email')
+                    ))
+                print("Subject: {subject}".format(subject=settings.FORUM_DIGEST_EMAIL_SUBJECT))
+                print text
+                print "=============================="
+            return
```

Comment:
This `return` at the very end is a bit tricky. There would be less chance of confusion if we moved the `do_forums_digests_flagged` call up to the top of the `handle` method, instead of tucking it in at the bottom.

Better yet, if we factor out this dry-run logic into its own method, things would be even more clear:

```python
if is_dry_run:
    do_forums_digests_flagged_dry_run()
else:
    do_forums_digests_flagged()
```